src/cms-content/vaccines/email-alerts.ts

```diff
@@ -0,0 +1,94 @@
+import html from 'remark-html';
+import remark from 'remark';
+import regions from 'common/regions';
+import { assert } from 'common/utils';
+import { getVaccinationDataByRegion } from './index';
+import { getVaccineInfoByFips, RegionPhaseGroup } from './phases';
+
+interface EmailPhaseInfo {
+  title: string;
+  subtitle: string;
+  description: string;
+  isMostRecentPhase: boolean;
+}
+
+export interface VaccinationEmailAlertData {
+  emailSubjectLine: string;
+  title: string;
+  subtitle: string;
+  sourceUrl: string;
+  sourceName: string;
+  eligibilityUrl: string;
+  vaccinationSignupUrl?: string;
+  currentPhases: EmailPhaseInfo[];
+}
+
+function markdownToHtml(markdownContent: string): string {
+  return remark().use(html).processSync(markdownContent).toString();
+}
+
+function formatVaccinationPhaseInfo(
+  phaseInfo: RegionPhaseGroup,
+  isMostRecentPhase: boolean,
+): EmailPhaseInfo {
+  const title = phaseInfo.tier
+    ? `${phaseInfo.phase}, ${phaseInfo.tier}`
+    : phaseInfo.phase;
+
+  return {
+    title,
+    subtitle: isMostRecentPhase ? 'most recently eligible' : 'eligible',
+    description: markdownToHtml(phaseInfo.description),
+    isMostRecentPhase,
+  };
+}
+
+/**
+ * Gathers vaccination information from the CMS and formats it to match what we need
+ * to pass to the email template in
+ * `scripts/vaccination_alerts/vaccination-alert-template.html`
+ */
+export function getEmailAlertData(fipsCode: string): VaccinationEmailAlertData {
+  const region = regions.findByFipsCodeStrict(fipsCode);
+
+  const vacinationInfo = getVaccineInfoByFips(fipsCode);
+  const vaccinationLinks = getVaccinationDataByRegion(region);
+
+  assert(
+    vacinationInfo,
+    `Vaccination info for FIPS ${fipsCode} not found in the CMS`,
+  );
+  assert(
+    vaccinationLinks?.eligibilityInfoUrl,
+    `Vaccination eligibility info for ${fipsCode} not found in the CMS`,
+  );
+
+  const currentPhases = vacinationInfo.phaseGroups
+    .filter(phaseInfo => phaseInfo.currentlyEligible)
+    .map((phaseInfo, index, allCurrentPhases) => {
+      const isLastCurrentPhase = index === allCurrentPhases.length - 1;
+      return formatVaccinationPhaseInfo(phaseInfo, isLastCurrentPhase);
+    });
+
+  const mostRecentPhase = currentPhases.find(
+    phaseInfo => phaseInfo.isMostRecentPhase,
+  );
+
+  const title = mostRecentPhase?.title
+    ? `${region.fullName} is now in ${mostRecentPhase.title} of vaccination`
```

Comment:
Does this mean if the eligibility groups for an existing phase changed, we should set `isMostRecentPhase` to `false` so that the email says "has updated vaccination information" instead of sending a redundant email with "is now in <existing phase> of vaccination" ?

Or do we not care?

I don't have strong feelings, but wondering if we should have a conversation or document expectations in a "How to update the CMS vaccine alert info" playbook somewhere or something.

Feel free to punt if I'm confused / over-thinking this.

Reply:
We are manually marking updates on the CMS vaccination information, but as a whole for the state, not on individual phases. Here, I'm assuming that we only do that when a vaccination phase starts or a new group is included in the "most recent" active phase, and that we are not emailing people if phases that started before update their definitions (I'm not sure if that happens).

The logic here is more of a fallback in case the `currentPhases` array is empty, but I guess in that case we should fail the job.